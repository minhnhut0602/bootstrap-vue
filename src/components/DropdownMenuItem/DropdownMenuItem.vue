<template>
    <li>
        <hr
            v-if="divider"
            class="dropdown-divider"
            :class="itemClass"
        >

        <h6
            v-else-if="header"
            class="dropdown-header"
            :class="itemClass"
        >
            <!-- @slot Displays content inside the dropdown menu item. -->
            <slot name="default">
                {{ label }}
            </slot>
        </h6>

        <Component
            v-else
            :is="is"
            class="dropdown-item"
            :class="[{active: active}, itemClass]"
            :href="href"
            :to="to"
            @click.prevent
        >
            <slot name="default">
                {{ label }}
            </slot>
        </Component>
    </li>
</template>

<script lang="ts" setup>
import {dropdownItemProps} from '@/composables/useDropdownItem';
import { PropType, computed } from 'vue';

const props = defineProps({
    /**
     * Adds the `active` class
     */
    active: {
        type: Boolean,
        default: false,
    },
    /**
     * Displays a divider
     */
    divider: {
        type: Boolean,
        default: false,
    },
    ...dropdownItemProps,
    /**
     * Displays label as header
     */
    header: {
        type: Boolean,
        default: false,
    },
    href: {
        type: String,
        default: '#',
    },
    is: {
        type: String,
        default: undefined,
    },
    /**
     * The label to display
     */
    label: {
        type: String,
        default: undefined,
    },
    to: {
        type: [Object, String] as PropType<string|Record<string, unknown>>,
        default: undefined,
    },
});

const is = computed(() => {
    if (props.is) {
        return props.is;
    }

    if (props.to) {
        return 'RouterLink';
    }

    return 'a';
});
</script>

<docs>
```vue
<DropdownMenuItem label="Menu item" />
```
</docs>
