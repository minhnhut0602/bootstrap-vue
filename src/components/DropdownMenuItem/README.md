# DropdownMenuItem

```vue live
<DropdownMenuItem />
```

## Props

| Prop name | Description | Type    | Values | Default   |
| --------- | ----------- | ------- | ------ | --------- |
| active    |             | boolean | -      | false     |
| divider   |             | boolean | -      | false     |
| header    |             | boolean | -      | false     |
| label     |             | string  | -      | undefined |
