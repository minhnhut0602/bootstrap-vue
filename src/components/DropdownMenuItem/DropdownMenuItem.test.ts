import DropdownMenuItem from '@/components/DropdownMenuItem';

describe('template', () => {
    componentSlotRenderTest(DropdownMenuItem);

    componentSlotRenderTest(
        DropdownMenuItem,
        'default',
        {props: {header: true}},
    );

    componentRenderTest(
        DropdownMenuItem,
        {props: {label: 'Hello'}},
        true,
        'renders label',
    );

    componentRenderTest(
        DropdownMenuItem,
        {props: {active: true, label: 'there'}},
        true,
        'renders active',
    );

    componentRenderTest(
        DropdownMenuItem,
        {props: {divider: true}},
        true,
        'renders divider',
    );

    componentRenderTest(
        DropdownMenuItem,
        {props: {header: true, label: 'general kenobi'}},
        true,
        'renders header',
    );
});
