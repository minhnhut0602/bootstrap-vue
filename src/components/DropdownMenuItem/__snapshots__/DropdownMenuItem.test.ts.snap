// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template renders active 1`] = `"<li><a class=\\"dropdown-item active\\" href=\\"#\\">there</a></li>"`;

exports[`template renders divider 1`] = `
"<li>
  <hr class=\\"dropdown-divider\\">
</li>"
`;

exports[`template renders header 1`] = `
"<li>
  <h6 class=\\"dropdown-header\\">
    <!-- @slot Displays content inside the dropdown menu item. -->general kenobi
  </h6>
</li>"
`;

exports[`template renders label 1`] = `"<li><a class=\\"dropdown-item\\" href=\\"#\\">Hello</a></li>"`;
